service-api/module/Application/src/Validators/LPAValidator.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Application\Validators;
+
+use Laminas\Validator\AbstractValidator;
+
+class LPAValidator extends AbstractValidator
+{
+    public const LPA = 'lpa';
+
+    protected array $messageTemplates = [
+        self::LPA => 'The LPA needs to be valid in the format M-XXXX-XXXX-XXXX',
+    ];
+
+    public function isValid($value): bool
+    {
+        $this->setValue($value);
+
+        return $this->lpaValidity($value);
+    }
+
+    private function lpaValidity(?string $lpa): bool
+    {
+        if ($lpa === null || $lpa === '') {
+            return true;
+        }
+            /** @var string $lpa */
+        if (1 == preg_match('/M(-([0-9A-Z]){4}){3}/', $lpa)) {
```

Comment:
Should use strict equality:
```suggestion
        if (false !== preg_match('/M(-([0-9A-Z]){4}){3}/', $lpa)) {
```

Reply:
Sorry, your solution `1 === preg_match()` is much more accurate because mine would have evaluated true for non-matches too 😅